File: scala/app/dto/ServerMove.scala
package dto

import java.util
import scala.collection.convert.wrapAll._

import play.api.libs.json.Json

object ServerMove {

  implicit val format = Json.format[ServerMove]

}

case class ServerMove(
  jumps: Map[String, Int],
  changes: Array[Array[Int]]
) {
  override def toString: String =
    "ServerMove{" +
    "jumps=" + jumps +
    ", changes=" + util.Arrays.deepToString(changes.asInstanceOf[Array[AnyRef]])  + "}"
}

File: java/src/main/java/com/anahoret/anadea_bot/dto/ServerMove.java
package com.anahoret.anadea_bot.dto;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Arrays;
import java.util.HashMap;

@Getter
@Setter
@NoArgsConstructor
public class ServerMove {

    private HashMap<String, Integer> jumps;
    private int[][] changes;

    @Override
    public String toString() {
        return "ServerMove{" +
                "jumps=" + jumps +
                ", changes=" + Arrays.deepToString(changes) +
                '}';
    }

}
